<!-- Launch iRobot handle ROS node, as well as our own iRobot hand control node -->
<launch>
  
  <arg name="hand_id" />
  <arg name="side" />
  
  <group ns="irobot_hands/$(arg side)_hand">
      <node pkg="handle_ros" type="handle_ros" name="handle_ros" output="screen" respawn="true">  
        <param name="hand_name" value="192.168.40.$(arg hand_id)" type="string"/>
        <param name="hand_port" value="3490" type="string"/>
        <param name="hand_udp" value="false" type="bool"/>
        <remap from="/handle/sensors/raw" to="/irobot_hands/$(arg side)_hand/sensors/raw"/>
        <remap from="/handle/control" to="/irobot_hands/$(arg side)_hand/control"/>
        <remap from="/handle/calibrate" to="/irobot_hands/$(arg side)_hand/calibrate"/>
      </node>
      
      <node pkg="mit_helios_scripts" type="mit_irobot_hand_control.py" name="mit_irobot_hand_control" args="$(arg side)" output="screen" respawn="false">
      </node>
  </group>
  
  

</launch>
